### PR TITLE
#300 - 4: Consultation duration calcaultion

### PR DESCRIPTION
**Pull request by Feng-Shui**
Re #300 

Details of the changes are in the commit messages.  Basically:
1. refactoring a few functions to make their params more explicit and new function to calculate inclusive days between two dates
2. Test coverage for these changes, unit tests for the specific functions and behat tests to cover their usage in templates.

Unit tests were for my sake as the code was refactored, I know their not in our testing pipeline, didn't make sense to throw them out though.

I'l going to add some more behat tests to cover a few more things soon.

Expecting to get a pass on this from GovBot...

### CODE REVIEW

File: modules/features/govcms_consultation/govcms_consultation.custom.inc [hunk 1]
```diff
@@ -218,21 +218,24 @@ function govcms_consultation_preprocess_field(&$variables, $hook) {
...
+    // processing below.
+    _govcms_consultation_consultation_vars($variables, $node);
     $consultation = $variables['consultation'];
 
-    // Return if formal submissions are not accepted.
-    if (!empty($consultation['hide_form'])) {
-      //field_group_hide_field_groups($variables['elements'], array('group_formal_submissions'));
+    // We hide the form if submissions are not enabled, or if the consultation
+    // period is not yet open or has closed.
+    if ($consultation['hide_form']) {
```
Comment (fiasco) [inline modules/features/govcms_consultation/govcms_consultation.custom.inc:238]: Why remove the `!empty()` checker which also does an implicit `isset` check?

**Comment by Feng-Shui** [comment]
Because `$consultation['hide_form']` is always set to a bool:

`$consultation['hide_form'] = !$consultation['submission_enabled'] || ($start > $now) || ($end < $now);`

**Comment by Feng-Shui** [comment]
@fiasco thoughts on this PR? Would prefer to branch the next PR from `govCMS:7.x-2.x` with this rolled in to keep merge conflicts down if possible.

**Comment by fiasco** [comment]
Sorry I've been absent here. Love the tests and am really keen to get them in but as I'm not very close to consultation, I'd like others to review. Perhaps @srowlands and any others who are interested in improving the consultation.

The main thing I'm looking to ensure is that any changes to existing functionality is bug fixes only and not accidentally introducing regressions.

**Pull request closed without merging**